src/igvfd/schemas/analysis_step.json

```diff
@@ -0,0 +1,138 @@
+{
+    "title": "Analysis Step",
+    "description": "Schema for submitting a step in a computational analysis workflow.",
+    "$id": "/profiles/analysis_step.json",
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "type": "object",
+    "required": [
+        "title",
+        "step_label",
+        "analysis_step_types",
+        "input_content_types",
+        "workflow"
+    ],
+    "identifyingProperties": [
+        "uuid",
+        "aliases"
+    ],
+    "additionalProperties": false,
+    "mixinProperties": [
+        {
+            "$ref": "mixins.json#/basic_item"
+        },
+        {
+            "$ref": "mixins.json#/attribution"
+        },
+        {
+            "$ref": "mixins.json#/standard_status"
+        }
+    ],
+    "dependentSchemas": {},
+    "properties": {
+        "schema_version": {
+            "default": "1"
+        },
+        "analysis_step_types": {
+            "title": "Analysis Step Types",
+            "description": "The classification of the software.",
+            "type": "array",
+            "uniqueItems": true,
+            "items": {
+                "title": "Type",
+                "type": "string",
+                "enum": [
+                    "alignment",
+                    "file format conversion",
+                    "signal generation"
+                ]
+            }
+        },
+        "step_label": {
+            "title": "Step Label",
+            "description": "Unique lowercased label of the analysis step using a naming system where the pipeline, software used, the purpose is described, e.g. rampage-grit-peak-calling-step",
```

Comment:
Should 'Pipeline' be 'Workflow'? Or maybe assay, as per the example?
I want to make sure if I should read this as a list of "possible options" for how a step might be named, or a suggestion that 3 things should be given in this order

Reply:
My interpretation was that it should include 3 items in that order. That being said, do we still need the pipeline's title if we are going to calculate it to generate the "name"?

Reply:
Hm. Maybe not required for greater uniqueness, but I think including the assay or similar "overall" term is nice for human readable context. It does make it longer, though. I can see a case either way, I leave it to your judgment